File: starfield/Star.js
class Star {
  constructor() {
    this.x = random(-width, width);
    this.y = random(-height, height);
    this.z = random(width);
    this.prevZ = this.z;
  }

  show() {
    fill(0, 200, 240, 300);
    noStroke();

    const circleSpreadX = map(this.x / this.z, 0, 1, 0, width);
    const circleSpreadY = map(this.y / this.z, 0, 1, 0, height);
    const diameter = map(this.z, 0, width, 10, 0);
    
    circle(circleSpreadX, circleSpreadY, diameter);

    const lineSpreadX = map(this.x / this.prevZ, 0, 1, 0, width);
    const lineSpreadY = map(this.y / this.prevZ, 0, 1, 0, height);

    this.prevZ = this.z;

    stroke(0, 0, 255, 100);
    line(lineSpreadX, lineSpreadY, circleSpreadX, circleSpreadY);
  }
  
    update() {
    this.z = this.z - speed;
      
    if (this.z < 1) {
      this.x = random(-width, width);
      this.y = random(-height, height);
      this.z = width;
      this.prevZ = this.z;
    }
  }
}

File: sketches/starfield/Star.py
class Star(object):
    def __init__(self):
        self.x = random(-width, width)
        self.y = random(-height, height)
        self.z = random(width)
        self.prev_z = self.z
    
    
    def show(self):
        fill(0, 200, 240, 300)
        noStroke()
    
        circle_spread_x = map(self.x / self.z, 0, 1, 0, width)
        circle_spread_y = map(self.y / self.z, 0, 1, 0, height)
        diameter = map(self.z, 0, width, 10, 0)
        
        circle(circle_spread_x, circle_spread_y, diameter)
    
        line_spread_x = map(self.x / self.prev_z, 0, 1, 0, width)
        line_spread_y = map(self.y / self.prev_z, 0, 1, 0, height)
    
        self.prev_z = self.z
    
        stroke(0, 0, 255, 100)
        line(line_spread_x, line_spread_y, circle_spread_x, circle_spread_y)
    
    
    def update(self, speed):
        self.z = self.z - speed
        
        if self.z < 1:
            self.x = random(-width, width)
            self.y = random(-height, height)
            self.z = width
            self.prev_z = self.z
            
